docs/en/sql_reference/data_types/nested_data_structures/simpleaggregatefunction.md

```diff
@@ -0,0 +1,36 @@
+# SimpleAggregateFunction(name, types\_of\_arguments…) {#data-type-simpleaggregatefunction}
+
+Unlike [`AggregateFunction`](../aggregatefunction.md), which stores not the value of the aggregate function but it's state:
+
+- `SimpleAggregateFunction` data type stores current value of aggregation and does not store aggregation state as [`AggregateFunction`](../aggregatefunction.md).
+.
+It supports simple stateless aggregate functions, including:
```

Comment:
```suggestion
- `SimpleAggregateFunction` data type stores current value of the aggregate function, and does not store its full state as [`AggregateFunction`](../aggregatefunction.md) does. This optimization can be applied to functions that are additive, in a following sense: the result of applying a function `f` to a row set `S1 + S2` can be obtained by applying `f` to parts of the row set separately, and then again applying `f` to the results: `f(S1 + S2) = f(f(S1) + f(S2))`. This property guarantees that  partial aggregation results are enough to compute the combined one, so we don't have to store and process any extra data.

Currently, the following aggregate functions are supported:
```

Reply:
But for example even simple `max` is not additive, no? (in the generic sense).

>`f(S1 + S2) = f(f(S1) + f(S2))`

Maybe it is better to use `,` over `+` here?

Reply:
Yep, it's additive in the sense of addition of row sets. Idk, maybe they have a proper word in relational algebra for this? And '+' is technically a multiset union -- UNION ALL or ⊎. Edited, but I think it's getting worse with each revision :joy:

Reply:
> I think it's getting worse with each revision

Well, actually now it looks better to me :+1: 

P.S. I think that if the type was named `StatelessAggregateFunction` it may help with understanding too (sigh).